system/autoware_system_monitor/src/cpu_monitor/cpu_monitor_base.cpp

```diff
@@ -229,167 +220,52 @@ void CPUMonitorBase::checkUsage()
 
-  // boost::process create file descriptor without O_CLOEXEC required for multithreading.
-  // So create file descriptor with O_CLOEXEC and pass it to boost::process.
-  int out_fd[2];
-  if (pipe2(out_fd, O_CLOEXEC) != 0) {
-    std::lock_guard<std::mutex> lock_snapshot(mutex_snapshot_);
-    usage_data_.clear();
-    usage_data_.summary_status = DiagStatus::ERROR;
-    usage_data_.summary_message = "pipe2 error";
-    usage_data_.elapsed_ms = 0.0f;
-    usage_data_.error_key = "pipe2";
-    usage_data_.error_value = strerror(errno);
-    return;
-  }
-  bp::pipe out_pipe{out_fd[0], out_fd[1]};
-  bp::ipstream is_out{std::move(out_pipe)};
-
-  int err_fd[2];
-  if (pipe2(err_fd, O_CLOEXEC) != 0) {
-    std::lock_guard<std::mutex> lock_snapshot(mutex_snapshot_);
-    usage_data_.clear();
-    usage_data_.summary_status = DiagStatus::ERROR;
-    usage_data_.summary_message = "pipe2 error";
-    usage_data_.elapsed_ms = 0.0f;
-    usage_data_.error_key = "pipe2";
-    usage_data_.error_value = strerror(errno);
-    return;
-  }
-  bp::pipe err_pipe{err_fd[0], err_fd[1]};
-  bp::ipstream is_err{std::move(err_pipe)};
+  std::lock_guard<std::mutex> lock_snapshot(mutex_snapshot_);
+  usage_data_.clear();  // The vector is cleared, but the allocated memory won't be reallocated.
 
-  int level = DiagStatus::OK;
   int whole_level = DiagStatus::OK;
-  std::vector<UsageData::CpuUsage> temporary_core_data{};
 
-  pt::ptree pt;
-  try {
-    // Execution of mpstat command takes 1 second.
-    // On failure, it will throw an exception or return non-zero exit code.
-    bp::child c("mpstat -P ALL 1 1 -o JSON", bp::std_out > is_out, bp::std_err > is_err);
-    c.wait();
-    if (c.exit_code() != 0) {
-      std::ostringstream os;
-      is_err >> os.rdbuf();
-      std::lock_guard<std::mutex> lock_snapshot(mutex_snapshot_);
-      usage_data_.clear();
-      usage_data_.summary_status = DiagStatus::ERROR;
-      usage_data_.summary_message = "mpstat error";
-      usage_data_.elapsed_ms = 0.0f;
-      usage_data_.error_key = "mpstat";
-      usage_data_.error_value = os.str();
-      return;
+  int32_t num_cores = cpu_usage_statistics_.get_num_cores();
+  for (int32_t core_index = 0; core_index < num_cores; ++core_index) {
+    CpuUsageStatistics::CoreUsageInfo usage = cpu_usage_statistics_.get_core_usage_info(core_index);
+    UsageData::CpuUsage cpu_usage;
+    cpu_usage.label = usage.name;
+    cpu_usage.usr = usage.user_percent;
+    cpu_usage.nice = usage.nice_percent;
+    cpu_usage.sys = usage.system_percent;
+    cpu_usage.iowait = usage.iowait_percent;
+    cpu_usage.idle = usage.idle_percent;
+    cpu_usage.total = usage.total_usage_percent;
+    // Some of the members of CpuUsageStatistics::CoreUsageInfo are not used.
+
+    // CpuUsageToLevel() converts usage to warning and error levels.
+    // It also counts occurrence of warning and error in addition to conversion.
+    float total_usage = usage.total_usage_percent * 1e-2;
```

Comment:
```suggestion
    const float total_usage = usage.total_usage_percent * 1e-2;
```
Consider applying this change if `total_usage` is not reassigned within `CpuUsageToLevel`.